ops/_private/timeconv.py

```diff
@@ -57,3 +72,40 @@ def parse_rfc3339(s: str) -> datetime.datetime:
 
     return datetime.datetime(int(y), int(m), int(d), int(hh), int(mm), int(ss),
                              microsecond=microsecond, tzinfo=tz)
+
+
+def parse_duration(s: str) -> datetime.timedelta:
+    """Parse a formatted Go duration.
+
+    This is similar to Go's time.ParseDuration function: it parses the output
+    of Go's time.Duration.String method, for example "72h3m0.5s". Units are
+    required after each number part, and valid units are "ns", "us", "µs",
+    "ms", "s", "m", and "h".
+    """
+    negative = False
+    if s and s[0] in '+-':
+        negative = s[0] == '-'
+        s = s[1:]
+
+    if s == '0':  # no unit is only okay for "0", "+0", and "-0"
+        return datetime.timedelta(seconds=0)
+
+    matches = list(_DURATION_RE.finditer(s))
+    if not matches:
+        raise ValueError('invalid duration: no number-unit groups')
+    if matches[0].start() != 0 or matches[-1].end() != len(s):
+        raise ValueError('invalid duration: extra input at start or end')
+
+    seconds = 0
+    for match in matches:
+        number, unit = match.groups()
+        if unit not in _DURATION_UNITS:
+            raise ValueError(f'invalid duration: invalid unit {unit!r}')
+        try:
+            seconds += float(number) * _DURATION_UNITS[unit]
+        except ValueError:
+            # Same exception type, but a slightly more specific error message
+            raise ValueError(f'invalid duration: {number!r} is not a valid float') from None
+
+    duration = datetime.timedelta(seconds=seconds)
```

Comment:
```
datetime.timedelta(days=0, seconds=0, microseconds=0, milliseconds=0, minutes=0, hours=0, weeks=0)
```

Reply:
I've taken this approach in https://github.com/canonical/operator/pull/1086/commits/40357ef11a51e4e0e4d501fabee7238189678d3d -- thanks